lib/Sema/TypeCheckType.cpp

```diff
@@ -4975,8 +4975,29 @@ TypeResolver::resolveSendingTypeRepr(SendingTypeRepr *repr,
     return ErrorType::get(getASTContext());
   }
 
+  NeverNullType resolvedType = resolveType(repr->getBase(), options);
+  
+  // If resolved type is Sendable, warn about unnecessary 'sending' annotation.
+  if (options.is(TypeResolverContext::FunctionInput)) {
+    if(!resolvedType->hasTypeParameter()) {
```

Comment:
I think we still want to check this if the type has a type parameter. If we're in the interface stage of type resolution, we can check the generic signature to see if it has a `Sendable` requirement.

Reply:
Updated the PR using generic signature to check for `Sendable` requirement